### PR TITLE
Add python3 support

### PR DESCRIPTION
**Pull request by garrettheel**

### CODE REVIEW

**Comment by ryan-lane** [comment]
Going to do some manual testing with this, but it looks like all the tricky bits are covered.

**Review by jacohend — APPROVED** [review]

File: confidant_client/__init__.py [hunk 1]
```diff
@@ -110,7 +120,7 @@ def __init__(
             'backoff': backoff,
             'assume_role': assume_role
         }
-        for key, val in args_config.iteritems():
+        for key, val in six.iteritems(args_config):
```
Comment (asottile) [inline confidant_client/__init__.py]: just use `.items()`, the "performance benefits" of `six.iter*` are not worth the code smell:

http://lists.openstack.org/pipermail/openstack-dev/2015-June/066391.html

File: confidant_client/__init__.py [hunk 2]
```diff
@@ -468,7 +482,7 @@ def create_blind_credential(
             'enabled': enabled
         }
         if store_keys:
-            data['credential_keys'] = credential_pairs.keys()
+            data['credential_keys'] = list(credential_pairs.keys())
```
Comment (asottile) [inline confidant_client/__init__.py:487]: or [`tuple(...)`](https://github.com/asottile/scratch/wiki/Prefer-tuples-to-lists) shrugs

File: confidant_client/cli.py [hunk 3]
```diff
@@ -457,7 +459,7 @@ class _HelpAction(argparse._HelpAction):
 
     def __call__(self, parser, namespace, values, option_string=None):
         parser.print_help()
-        print ''
+        print('')
```
Comment (asottile) [inline confidant_client/cli.py:462]: `print()` is equivalent to `print('')`

(also `print` is equivalent to `print ''` in python2)

File: confidant_client/cli.py [hunk 4]
```diff
@@ -466,10 +468,10 @@ def __call__(self, parser, namespace, values, option_string=None):
         for subparsers_action in subparsers_actions:
             # get all subparsers and print help
             for choice, subparser in subparsers_action.choices.items():
-                print ('Subcommand \'{0}\':'.format(choice))
-                print (subparser.format_help())
+                print('Subcommand \'{0}\':'.format(choice))
+                print(subparser.format_help())
```
Comment (asottile) [inline confidant_client/cli.py]: there's a `subparser.print_help()`

File: confidant_client/__init__.py (hunk 1 above)
Comment (garrettheel) [reply confidant_client/__init__.py]: All of these nits are on the output of `modernize` or existing code. I can fix some of them for now, but I'm not sure the effort is really worth it

Comment (asottile) [reply confidant_client/__init__.py]: I was hoping you would take this advice in the rest of the `six.iter*` calls :)

File: confidant_client/__init__.py [hunk 5]
```diff
@@ -430,9 +440,11 @@ def _get_keys_and_encrypted_pairs(
...
                 _kms
             )
-            data_keys[region] = base64.b64encode(
-                ensure_bytes(data_key['ciphertext'])
-            ).decode('ascii')
+            data_keys[region] = ensure_text(
+                base64.b64encode(
+                    ensure_bytes(data_key['ciphertext'])
+                )
+            )
```
Comment (asottile) [inline confidant_client/__init__.py]: `b64encode` is *always* going to return `bytes` so it should use `.decode('ASCII')` and not `ensure_text` which guesses.

Comment (asottile) [reply confidant_client/__init__.py]: I think the same is true for `f.encrypt()` below, but I'm not familiar with that API

**Review by ryan-lane — APPROVED** [review]

**Pull request merged**